snapcraft/common.py

```diff
@@ -61,21 +61,40 @@ def run_output(cmd, **kwargs):
                                        **kwargs).decode('utf8').strip()
 
 
+_DEB_ARCH = {
+    'armv7l': 'armhf',
+    'aarch64': 'arm64',
+    'i686': 'i386',
+    'ppc64le': 'ppc64el',
+    'x86_64': 'amd64',
+}
+
+
 def get_arch():
-    global _arch
-    if _arch is None:
-        _arch = subprocess.check_output(
-            ['dpkg-architecture', '-qDEB_BUILD_ARCH']).decode('utf8').strip()
-    return _arch
+    try:
+        return _DEB_ARCH[platform.machine()]
+    except KeyError:
+        raise EnvironmentError(
+            '{} is not supported, please log a bug at'
+            'https://bugs.launchpad.net/snapcraft'.format(platform.machine()))
+
+
+_GNU_ARCH_TRIPLET = {
+    'armv7l': 'arm-linux-gnueabihf',
+    'aarch64': 'aarch64-linux-gnu',
+    'i686': 'i386-linux-gnu',
+    'ppc64le': 'powerpc64le-linux-gnu',
+    'x86_64': 'x86_64-linux-gnu',
+}
```

Comment:
This information is very closely related to the `_DEB_ARCH` above. Would the codebase be less prone to an update in one spot and forgetting to update the other if you stored this information within the same container? For example, a nested map of the structure:

```python
{
   'armv7l': {
      'arch': 'armhf',
      'triplet': 'arm-linux-gnueabihf',
   }
   # ...
}
```

You get the idea. Less duplication, still clear?